src/store/user/reducer.ts

```diff
@@ -1,26 +1,36 @@
 import { TUser } from 'src/shared.types';
-import { Login, Logout } from './actions';
+import { GetUser, Login, Logout } from './actions';
 import { createReducer } from '@reduxjs/toolkit';
 
 const initUsersState = {
-	value: JSON.parse(localStorage.getItem('user')),
-} as { value: TUser };
+	value: {
+		email: '',
+		name: '',
+		token: '',
+		role: '',
+		isAuth: false,
+	},
+	isLoaded: false,
+} as { value: TUser | string[]; isLoaded: boolean };
 
 const usersReducer = createReducer(initUsersState, (builder) => {
 	builder
-		.addCase(Login, (state, action) => {
-			localStorage.setItem('user', JSON.stringify(action.payload));
-			state.value = action.payload;
+		.addCase(Login.fulfilled, (state, action) => {
+			const user = { ...action.payload, isAuth: true };
+			localStorage.setItem('token', (action.payload as TUser).token);
+			state.value = user;
 		})
-		.addCase(Logout, (state) => {
-			const logoutUser: TUser = {
-				email: '',
-				name: '',
-				token: '',
-				isAuth: false,
+		.addCase(Logout.fulfilled, (state) => {
+			localStorage.removeItem('token');
+			state.value = { ...initUsersState.value };
+			state.isLoaded = false;
+		})
+		.addCase(GetUser.fulfilled, (state: typeof initUsersState, payload) => {
+			state.isLoaded = true;
+			state.value = {
+				...payload.payload,
+				token: localStorage.getItem('token'),
```

Comment:
no need to retrieve the token from localStorage here. It should already be part of the state, so it's better to rewrite:
```
state.value = {
   ...state.value,
   ...payload.payload,
}
```